x-pack/plugins/serverless_security/public/components/get_started/use_storage.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useMemo } from 'react';
+import type { Storage } from '@kbn/kibana-utils-plugin/public';
+import { CardId, ProductId, StepId } from './types';
+
+export const ACTIVE_PRODUCTS_STORAGE_KEY = 'ACTIVE_PRODUCTS';
+export const FINISHED_STEPS_STORAGE_KEY = 'FINISHED_STEPS';
+
+export const useStorage = (storage: Storage) => {
+  const addFinishedStepToStorage = useCallback(
+    (cardId: CardId, stepId: StepId) => {
+      const finishedSteps: Record<CardId, Record<StepId, boolean> | undefined> =
+        storage.get(FINISHED_STEPS_STORAGE_KEY) ?? {};
+      const card: Record<StepId, boolean> =
+        finishedSteps[cardId] ?? ({} as Record<StepId, boolean>);
+      if (!card[stepId]) {
+        card[stepId] = true;
```

Comment:
Are we ever going to have a `false` value here? If the answer is no I think it would make more sense to use: 

`const finishedSteps: Partial<Record<CardId, StepId[]>>`

it will simplify the logic outside as well